providers/mlx5/dr_domain.c

```diff
@@ -344,7 +344,7 @@ mlx5dv_dr_domain_create(struct ibv_context *ctx,
 
 	ret = pthread_spin_init(&dmn->info.rx.lock, PTHREAD_PROCESS_PRIVATE);
 	if (!ret)
-		pthread_spin_init(&dmn->info.tx.lock, PTHREAD_PROCESS_PRIVATE);
+		ret = pthread_spin_init(&dmn->info.tx.lock, PTHREAD_PROCESS_PRIVATE);
```

Comment:
Shouldn't pthread_spin_destroy() be called here? (on both locks tx and rx).

Reply:
Thanks Mark, the patch was updated to handle that.